src/modules/user/user.usecase.spec.ts

```diff
@@ -4132,6 +4139,100 @@ describe('User use cases', () => {
       );
       expect(mockMailLimit.increaseTodayAttempts).not.toHaveBeenCalled();
     });
+
+    it('When Klaviyo tracking fails, then should still send email successfully', async () => {
+      const mockMailLimit = newMailLimit({
+        userId: mockUser.id,
+        attemptsCount: 1,
+        attemptsLimit: 5,
+      });
+      jest
+        .spyOn(mockMailLimit, 'isLimitForTodayReached')
+        .mockReturnValue(false);
+      jest.spyOn(mockMailLimit, 'increaseTodayAttempts');
+
+      jest
+        .spyOn(mailLimitRepository, 'findOrCreate')
+        .mockResolvedValue([mockMailLimit, true]);
+      jest
+        .spyOn(mailLimitRepository, 'updateByUserIdAndMailType')
+        .mockResolvedValue(undefined);
+      jest
+        .spyOn(mailerService, 'sendIncompleteCheckoutEmail')
+        .mockResolvedValue(undefined);
+
+      const klaviyoError = new Error('Klaviyo API error');
+      jest
+        .spyOn(klaviyoService, 'trackCheckoutStarted')
+        .mockRejectedValue(klaviyoError);
+
+      const result = await userUseCases.handleIncompleteCheckoutEvent(
+        mockUser,
+        mockIncompleteCheckoutDto,
+      );
+
+      expect(result).toEqual({ success: true });
+      expect(klaviyoService.trackCheckoutStarted).toHaveBeenCalledWith(
+        mockUser.email,
+        {
+          checkoutUrl: mockIncompleteCheckoutDto.completeCheckoutUrl,
+          planName: mockIncompleteCheckoutDto.planName,
+          price: mockIncompleteCheckoutDto.price,
+        },
+      );
+      expect(mailerService.sendIncompleteCheckoutEmail).toHaveBeenCalledWith(
+        mockUser.email,
+        mockIncompleteCheckoutDto.completeCheckoutUrl,
+      );
+      expect(mockMailLimit.increaseTodayAttempts).toHaveBeenCalled();
+      expect(loggerMock.error).toHaveBeenCalledWith(
+        expect.stringContaining('Failed to track checkout in Klaviyo'),
+      );
+    });
+
+    it('When Klaviyo tracking succeeds, then should track event with correct data', async () => {
```

Comment:
This test should be related to sending the email as these are the user's use case tests. If you want to test the tracking service, create a file for that